### ISSUE TITLE
How to create a continuous enviornment

### ISSUE THREAD

**Issue by Bhaney44** [tensorflow/agents#145]
One problem I am running into is developing an environment for an agent to solve. Generally, in RL the environment represents the problem for the agent to solve. My impression is with tensor-flow the environment is made up of tensor matrices for each state. However, I have been unsuccessful in figuring out how to create an environment that evolves with time - where an agent can learn to optimize a specific reward.

One problem with creating an environment is whether to use sample data. Presumably, the environment is made up of a sequence of states - the value of which is predicted by a neural network. I would sincerely appreciate any suggestions or advice on how to create an environment.

With thanks,

Brian

**Comment by kbanoop** [comment]
The easiest way would be to write an environment in python as shown in this colab: 
https://github.com/tensorflow/agents/blob/master/tf_agents/colabs/2_environments_tutorial.ipynb

Once you have that, you can use the TFPyEnvironment wrapper to move it into TensorFlow:
https://github.com/tensorflow/agents/blob/master/tf_agents/environments/tf_py_environment.py

Writing an environment directly in TensorFlow is a little harder and would require using a tf.Variable to keep track of the state, and updating this every time you call environment.step(). e.g. see:
https://github.com/tensorflow/agents/blob/master/tf_agents/environments/tf_environment_test.py

**Comment by Bhaney44** [comment]
That's super helpful. Thanks!
